File: 0896-monotonic-array/0896-monotonic-array.py
class Solution:
  def isMonotonic(self, nums: List[int]) -> bool:
    inc = None
    for i in range(1, len(nums)):
      if inc == None:
        if nums[i] > nums[i - 1]:
          inc = True
        elif nums[i] < nums[i - 1]:
          inc = False
      elif nums[i] < nums[i - 1] if inc else nums[i] > nums[i - 1]:
        return False
    return True

File: 0896-monotonic-array/0896-monotonic-array.ts
function isMonotonic(nums: number[]): boolean {
  let inc = true, dec = true;
  for (let i = 1; i < nums.length; i++) {
    if (inc && nums[i] < nums[i - 1]) inc = false;
    if (dec && nums[i] > nums[i - 1]) dec = false;
    if (!inc && !dec) return false;
  }
  return true;
};